manifests/user.pp

```diff
@@ -74,9 +74,22 @@
       /usr/bin/ipmitool user test ${user_id} 16 "\$PASSWORD" || \
       /usr/bin/ipmitool user test ${user_id} 20 "\$PASSWORD"
       |- CMD
+
+    # Password capacity parameter defaults to 16 if not provided
+    #  and will result in truncated passwords
+    if $real_password.length <= 16 {
+      $password_capacity = '16'
+    }
+    if $real_password.length > 16 and $real_password.length <= 20 {
+      $password_capacity = '20'
+    }
+    if $real_password.length > 20 {
```

Comment:
If this conditional is evaluated first, the value of `$real_password.length` would always be `<= 20`.  That would allow one of the conditional statements to be removed. E.g. 
```
  if $real_password.length > 20 {
    fail('ipmi v2 restricts passwords to 20 or fewer characters')
  }
  if $real_password.length <= 16 {
    $password_capacity = '16'
  } else {
    $password_capacity = '20'
  }
```

Reply:
Good call. I also re-ordered such that the unless_command is right next to the place it is used. I've squashed these commits together to tidy this up.